tests/coverage/unreachable/variant/main.rs

```diff
@@ -12,6 +12,7 @@ enum Direction {
 }
 
 fn print_direction(dir: Direction) {
+    // For some reason, `dir`'s span is reported as `UNCOVERED` too
```

Comment:
This is weird. Any insights why the span is reported as uncovered?

Reply:
No insights, unfortunately. I'm going to ask around because it's really weird. For the time being, I've created #3456 and pasted it into the test.